### ISSUE TITLE
Notifications no longer work (and I've tried the steps on the Troubleshooting page)

### ISSUE THREAD

**Issue by ryancwalsh** [signalapp/Signal-Android#12136]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I don't receive notifications anymore. I've used Signal for years (and love it!) and have never had this problem. 

Whether I'm on wifi or cellular, I don't get Signal notifications. 

I don't know whether it's relevant, but I think it started happening recently when I traveled to Turkey and inserted a Turkey sim card. I figured maybe that had something to do with it and that the problem wouldn't persist when I returned to USA and swapped back to my original sim card. But the problem persists. The sim card swap could be a red herring, but I figured I'd mention it.

I've already tried the steps at https://support.signal.org/hc/en-us/articles/360007318711-Troubleshooting-Notifications
But I couldn't find this setting on my phone:
> Confirm you have enabled WiFi or cellular data while your phone is asleep. Android phone Settings > Wi-Fi > Advanced > Always keep Wi-Fi on during sleep.

And I haven't tried "Reset your Google Account." or completely uninstalling and reinstalling Signal (since I want to maintain my conversation history).

https://github.com/signalapp/Signal-Android/issues/12068 caught my eye, but I don't think it covers what's happening for me.

### Steps to reproduce
- Have a friend send a Signal message to me
- Notice that there is no Signal notification, yet there is an unread message in the Signal conversation when I open the Signal app

**Actual result:** I never see Signal notifications. I need to proactively enter the Signal app to check for messages.
**Expected result:** Signal should show notifications (assuming I leave the setting enabled), like it used to.

### Device info

**Device:** Google Pixel 3
**Android version:** 12
**Signal version:** 5.34.10

### Link to debug log

https://debuglogs.org/android/5.34.10/ae003e8fdbabdbb5748b6b94e9da6dbeb2c4e4b4d625b547aabc08445d47fad3

**Comment by cody-signal** [comment]
Thanks, we are having problems with Android 12 and notifications as others have reported this as well. We will be looking into this soon.

**Comment by kayeekay** [comment]
I have the same issue. My brother does not have the issue, though we're in the same house, with the same internet, same carrier, with the same model of phone (Pixel 6). The only difference I can tell in Signal is that I become a beta tester a few days ago...I retracted and tested, but no luck. I found if you create a custom sound for a contact (other than the default), the notification works. I'm not sure if it works 100% of the time.

**Comment by ryancwalsh** [comment]
Thanks for your response, @cody-signal .

Do you have any clues yet? 

I heard from my sister that her Signal notifications stopped working so long ago that she can't even remember what they were like.  And she said that she feels like occasionally something works for custom sounds for an individual contact (as @kayeekay  mentioned), but I've tried setting a custom notification for an individual contact, and it hasn't helped. 

Plus, I need the notifications to work for _all_ conversations, as they used to.

I'm available to help you or your team with any debugging; this is very important to me. I really like Signal and want it to thrive.

Thanks.

**Comment by ryancwalsh** [comment]
Update: yesterday (or _maybe_ even the day before?) and today my Signal (Android app) notifications have started working sporadically (for certain but not all conversations / contacts).

I don't think I'd made any changes to my phone or my app.

**Comment by ryancwalsh** [comment]
@cody-signal Hi Cody, my Signal notifications still aren't working except if I choose "Use custom notifications", and even then I think they don't always work.  So it's really making Signal difficult to rely on.  I'm not noticing messages until hours later, which makes everyday coordination (of meeting up in person, etc) difficult. 

You said _"others have reported this as well. We will be looking into this soon."_  I'd love to hear any update! Thanks!

**Comment by UserX404** [comment]
@ryancwalsh 
Would be nice if you could append fresh logs when you report a specific behavior and/or some changes.
This could help the devs to track down issues and isn't something difficult to do.

**Comment by cody-signal** [comment]
Fresh debuglogs from running the latest build would be appreciated.

**Comment by userj** [comment]
> Fresh debuglogs from running the latest build would be appreciated.

I'm a bit new to android debuglogs. Is there a method that you would like or a quick guide how to get it specifically for signal? Thanks

**Comment by userj** [comment]
Oh that was easy. Thank you for your help.
Here's the log:
https://debuglogs.org/android/5.41.11/7ad3f866b0b06e0fe715701b20ec1305e3f235a2e92096e935e30bdf41297509

**Comment by Doge7734** [comment]
I also sometimes don't receive messages on my Pixel 6 with stock Android running Android 12. For the most part it's fine though. I'll get a debug log the next time it happens. My Signal version is 5.41.13. After happening a few times it fixes itself, so it isn't constant but I don't believe I was in an area with poor connectivity.

**Comment by ryancwalsh** [comment]
This issue has continued for me so I recommend keeping it Open.

**Comment by jace** [comment]
This has been bugging me for months. Nothing I do seems to enable notifications on Signal. I'm on Android 12 too.

**Comment by jace** [comment]
Huh, I think I've found a fix courtesy an obscure Reddit comment.

Signal notifications had a ringtone that no longer exists. Changing the ringtone has enabled notifications again.

**Comment by ryancwalsh** [comment]
@jace That's really interesting. 

(https://github.com/signalapp/Signal-Android/issues/12136#issuecomment-1097424066 had mentioned custom notifications, so I wonder if that's somehow related.)

Do you have a link to that Reddit page or anything else that might give the Signal team even more of a clue?

But I bet this would be enough to tell them how to fix it anyway. They could just look at which the default ringtone is on Signal on Android 12 and probably edit the code to do a check first to ensure that that ringtone exists and fall back to one that definitely exists if the first one doesn't.

Right, @cody-signal ? 

Thanks!

**Comment by azuredusk10** [comment]
My friend on Android started having problems with Signal message notifications a few days ago. 

I’m on iOS and we’ve been texting through Signal with no issue for several months. Now, she no longer receives notifications when the app is closed, and when I send her messages, they only show up with one check mark until she manually opens Signal. This started happening on 25 October. 

Timestamps of sent messages which weren’t delivered until she opened the app:
1666716833172 
(25 Oct)

1666978198078
1666978205391
(Today, 28 Oct)

Here is her debug log, from today, after she opened Signal and then received the messages with the above timestamps:
https://debuglogs.org/android/5.52.5/7efd3f2c3551855fb8d14ad2116c72175925f75646f8f33d948682c66396294f

She says she didn’t change any settings recently, and she’s checked her phone’s notification settings and Signal has been allowed to send notifications.

**Comment by jace** [comment]
@azuredusk10 ask her to check the ringtone for the notification. I had the same problem - Signal was allowed to send notifications, but because it had been set to a sound that was no longer on my phone, the notification wasn't being shown.

**Comment by azuredusk10** [comment]
Further to my last comment, my contact resolved the issue by changing a setting in her phone which was previously set to “auto”. She didn’t specify which; I’m guessing it was something to do with power management for the Signal app.

**Comment by yarsiemanym** [comment]
My father an di are both having this issue. He is on iOS and I am on Android 13 (GrapheneOS). It worked flawlessly until a few weeks ago. Now neither of get push notifications any more. I have tried the standard checklist and also tried changing the notification sound as someone above recommended. No change.

**Comment by Rufusdotrufus** [comment]
> This comment: https://www.reddit.com/r/signal/comments/s84gk1/comment/iemkbrt/?utm_source=share&utm_medium=web2x&context=3

Wow what an easy fix for a problem that has been plaguing my signal app on Pixel 7.  I had tried almost everything under the sun to re-enable ringtone notifications.  **Something like that should be on a Signal help page regarding notifications not working.**

**Comment by yarsiemanym** [comment]
So for me this ended up being an issue with Google Play Services on GrapheneOS. for a while, it seemed like only Signal wasn't getting notifications because it wasn't giving me _any_, but then I noticed that other apps also weren't giving me all of the notifications they were supposed to either. Signal was the only app that was giving absolutely none, but other apps were spotty at times. It was really frustrating. Then one day, I walked into the grocery store, my phone joined the WiFi network there since I had used it previously, and all of a sudden I got flooded with notifications from all sorts of apps. Everything seems to be working fine since then, but it was really strange.

**Comment by optimismus** [comment]
Big thanks to @jace :-) it solved exactly my problem. I found it here: Signal App >> Settings >> Notification >> Chat >> Configure (? mine is German: anpassen) >> Ringtone
It was on "Standard", but the first "Standard" in the list, I changed to the second "Standard" in the list and it works like a charm :-)

@ryancwalsh **As already said: It should be in dhe FAQ or signal help page!**

**Comment by rondonjon** [comment]
> Big thanks to @jace :-) it solved exactly my problem. I found it here: Signal App >> Settings >> Notification >> Chat >> Configure (? mine is German: anpassen) >> Ringtone It was on "Standard", but the first "Standard" in the list, I changed to the second "Standard" in the list and it works like a charm :-)
> 
> @ryancwalsh **As already said: It should be in dhe FAQ or signal help page!**

If this is really the reason why thousands (and possibly many more) users no longer receive notifications and the Signal app has thus effectively become useless for them, then this should not only be explained in some FAQ.

**The settings have to be checked in the app the next time it is started and, if necessary, changed automatically!**

In my circle of friends, several users have been complaining for about a year that they no longer receive notifications. For the average user, this makes the app a candidate for deletion. Very few ordinary users are even able to check the app settings. Let alone search the Internet for FAQs and find a solution somewhere.

**Comment by HensonHorkson** [comment]
> If this is really the reason why thousands (and possibly many more) users no longer receive notifications and the Signal app has thus effectively become useless for them, then this should not only be explained in some FAQ.
> 
> **The settings have to be checked in the app the next time it is started and, if necessary, changed automatically!**
> 
> In my circle of friends, several users have been complaining for about a year that they no longer receive notifications. For the average user, this makes the app a candidate for deletion. Very few ordinary users are even able to check the app settings. Let alone search the Internet for FAQs and find a solution somewhere.

Same with me, with many friends we decided to not use Signal anymore as there is often no notification, nor on screen, nor the dot next to the icon, nor vibration. Sometimes it works, sometimes we only see it, when we open the app. On my phone "pixie dust" was defaultly selected as sound, which all other messengers can use. 

It really pains my heart to only use it with small dwingling small group of people. It even hurts more that I can´t recommend it anymore for the trouble it has caused.

**Comment by cody-signal** [comment]
@rondonjon @HensonHorkson Which device manufacturers have y'all experienced the issue on or believe others have? Samsung? OnePlus? etc. OS versions would be helpful to know as well.

**Comment by HensonHorkson** [comment]
> [@rondonjon](https://github.com/rondonjon) [@HensonHorkson](https://github.com/HensonHorkson) Which device manufacturers have y'all experienced the issue on or believe others have? Samsung? OnePlus? etc. OS versions would be helpful to know as well.

@cody-signal thanks for coming back to me! 

I tested it today with 12 contacts. Everything worked out, everybody received push messages and notifications. 

Couldn´t check it with the guy with whom it happend most frequently. He has/had an about three years old Android phone, but I´m not certain. 

I hope the issue was solved with one of the latest updates.

If it helps you to track the issue: I´m using a Shift6mq with Android 10 and ShiftOS 3.10 G.

Thank you very much for your message, otherwise I wouldn´t have testet it again.

**Comment by akorn** [comment]
> [@rondonjon](https://github.com/rondonjon) [@HensonHorkson](https://github.com/HensonHorkson) Which device manufacturers have y'all experienced the issue on or believe others have? Samsung? OnePlus? etc. OS versions would be helpful to know as well.

I experienced it on a Google Pixel 6 Pro, when I had an overzealous DNS based ad and tracker blocker that blocked firebaseinstallations.googleapis.com. Unblocking that after Signal initially failed to register with it didn't help, because FCM registration wasn't retried.

I believe this was a bug that got fixed about a year ago and Signal now retries registration occasionally?

At about that time I switched to Molly, so my current experience may not be relevant, but notifications are now getting delayed again (maybe since mid-January or so). I haven't tried any troubleshooting yet.

**Comment by ryancwalsh** [comment]
Interesting point. I have a Google Pixel 6 and use the free version of https://one.one.one.one/ continually.

I wonder if that's why sometimes Signal voice calls don't work (I answer someone's calls but hear only silence) and why sometimes notifications don't work or are delayed. 

But if 1.1.1.1 is related to these problems, I don't know why it would only _sometimes_ happen.

**Comment by cody-signal** [comment]
> I tested it today with 12 contacts. Everything worked out, everybody - including me - received push messages and notifications.

Cool. Thanks for the details. I don't have that device so can't  test the missing sound causing missing notification issue, and the devices I have tried it on it works. The OS either gives the default sound, or nothing, but the notification still shows.

> I believe this was a bug that got fixed about a year ago and Signal now retries registration occasionally?

It's a tricky one as folks have different use cases and patterns around fcm.

> But if 1.1.1.1 is related to these problems, I don't know why it would only sometimes happen.

I doubt 1.1.1.1 is causing these issues. If you do run into calling issues, please feel free to open a new issue with a debuglog and we can look into the separately. If you are actively encountering notification issues though, please provide a new debuglog so we can see if anything odd shows up. Please and thank you.

**Comment by HensonHorkson** [comment]
> Cool. Thanks for the details. I don't have that device so can't test the missing sound causing missing notification issue, and the devices I have tried it on it works. The OS either gives the default sound, or nothing, but the notification still shows.

Thanks COdy. During the last days with my shiftphone it worked with custom sounds as well.

**Comment by ThornedRose** [comment]
I'm having intermittent reliability with Signal notifications. I am however using a Google Pixel 7 with GrapheneOS and I realise that Graphene itself can cause issues if it's not correctly set up. However, I have read all relevant documentation from both GrapheneOS and Signal and I believe that everything is set up as it should.
This is on a fresh phone (factory reset, updated to latest version of Android) and then GrapheneOS installed.
I initially installed Signal via the apk file. However, in troubleshooting the lack of notifications, I uninstalled Signal and then reinstalled it via Google Play Store (sandboxed since it's GrapheneOS). Google Play Services has permissions for network and notifications, etc. That seems to have improved the reliability but it's still pot luck as to whether I get notifications or not.
It seems like Signal is being 'slept' even though all the settings allow it to run in the background, etc. Sometimes when changing between wi-fi and mobile data, that will trigger notifications to come through. As will restarting my phone and then unlocking it. Otherwise I can find no particular pattern as to when I get notifications and when I don't. :/
Again, I am willing to concede I have done something wrong since on my previous Samsung Galaxy S9+ (Android 10 I think it is), notifications were always almost instant. :/
